amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,330 @@
+                                color='tab:red', alpha=0.3)
+
+            if isinstance(components['large']['mean'], np.ndarray):
+                ax.plot(x, components['large']['mean'], color='tab:green', label='large')
+                ax.fill_between(x, components['large']['mean'] - components['large']['std'],
+                                components['large']['mean'] + components['large']['std'],
+                                color='tab:green', alpha=0.3)
+
+        elif keywords.get('composition'):
+            ptype = 'composition'
+            if isinstance(components['pure']['mean'], np.ndarray):
+                ax.plot(x, components['pure']['mean'], color='tab:red', label='pure')
+                ax.fill_between(x, components['pure']['mean'] - components['pure']['std'],
+                                components['pure']['mean'] + components['pure']['std'],
+                                color='tab:red', alpha=0.3)
+
+            if isinstance(components['nitrogen']['mean'], np.ndarray):
+                ax.plot(x, components['nitrogen']['mean'], color='tab:green', label='nitrogen')
+                ax.fill_between(x, components['nitrogen']['mean'] - components['nitrogen']['std'],
+                                components['nitrogen']['mean'] + components['nitrogen']['std'],
+                                color='tab:green', alpha=0.3)
+
+        else:
+            ptype = None
+
+        ax.axhline(0, linestyle='--', color='gray')
+        ax.legend(fontsize=10)
+
+        if keywords.get('save'):
+            if not isinstance(ptype, str):
+                ptype = 'fitted'
+            fig.savefig(f"{keywords['save']}mc_{ptype}_breakdown.pdf")
+        else:
+            plt.show()
+
+        plt.close(fig)
+
+    def write(self, filename: str = "") -> None:
+        """
+        Write the spectra to file as an IPAC-table.
+
+        """
+        import sys
+        import datetime
+        from astropy.table import Table  # type: ignore
+
+        if not filename:
+            filename = self.__class__.__name__ + ".tbl"
+
+        hdr = list()
+
+        kv = {
+            "DATE": datetime.datetime.now()
+            .astimezone()
+            .replace(microsecond=0)
+            .isoformat(),
+            "ORIGIN": "NASA Ames Research Center",
+            "CREATOR": f"Python {sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}",
+            "SOFTWARE": "AmesPAHdbPythonSuite",
+            "AUTHOR": "Dr. C. Boersma",
+            "TYPE": self.__class__.__name__.upper(),
+            "SAMPLES": f'{len(self.mcfits)}'
+        }
+
+        for key, value in kv.items():
+            if not value.isnumeric():
+                hdr.append(f"{key:8} = '{value}'")
+            else:
+                hdr.append(f"{key:8} = {value}")
+
+        tbl = Table(names=('attribute', 'mean', 'std', 'skew', 'kurt'),
+                    dtype=('U25', 'float64', 'float64', 'float64', 'float64',),
```

Comment:
Is there a way not to hard-code dtype?

Reply:
I believe only when importing astropy.ascii along with Table.